File: dynamo_go/lambda/orders/cancel/cancelOrder.go
package main

import (
	"github.com/aws/aws-lambda-go/events"
	"github.com/aws/aws-lambda-go/lambda"
	"github.com/aws/aws-sdk-go/aws"
	"github.com/aws/aws-sdk-go/aws/session"
	"github.com/aws/aws-sdk-go/service/dynamodb"
	"os"
	"webshop/internal/apigw"
)

func Handler(event events.APIGatewayProxyRequest) (*events.APIGatewayProxyResponse, error) {
	ddb := dynamodb.New(session.New())

	_, err := ddb.UpdateItem(&dynamodb.UpdateItemInput{
		TableName: aws.String(os.Getenv("ORDERS_TABLE")), //aws.String macht wert zu einem  pointer
		Key: map[string]*dynamodb.AttributeValue{
			"orderNo": {S: aws.String(event.PathParameters["orderNo"])},
			"userId":  {S: aws.String(event.PathParameters["userId"])},
		},
		UpdateExpression: aws.String("set orderStatus=:s"),
		ExpressionAttributeValues: map[string]*dynamodb.AttributeValue{
			":s": {S: aws.String("Canceled")},
		},
		ReturnValues: aws.String("ALL_NEW"),
	})

	if err != nil {
		return apigw.CreateResponse(500, err.Error()), err
	}

	return apigw.CreateResponse(200, "Order Canceled"), err
}

func main() {
	lambda.Start(Handler)
}

File: monolith_ts/src/orders/cancelOrder.ts
import { DynamoDB } from 'aws-sdk';
import { HttpRequest, HttpResponse } from '../http'
import { OrderStatus } from './forms';


const ddb = new DynamoDB.DocumentClient({ region: "eu-central-1" })

export async function handler(event: HttpRequest): Promise<HttpResponse> {
  const { userId, orderNo } = event.pathParameters
  const params = {
    TableName: process.env.ORDERS_TABLE!,
    Key: {
      userId,
      orderNo
    },
    UpdateExpression: "set orderStatus=:s",
    ExpressionAttributeValues: {
      ":s": OrderStatus.Canceled,
    },
    ReturnValues: "ALL_NEW"
  }
  
  try {
    const data = await ddb.update(params).promise()
    if (data.Attributes) {
      return {
        statusCode: 200,
        headers: {
          'Access-Control-Allow-Origin': '*',
          'Access-Control-Allow-Credentials': true,
        },
        body: 'Order was canceled'
      }
    }
    return {
      statusCode: 404,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: "Item not found"
    }
  } catch (err) {
    return {
      statusCode: err.statusCode || 500,
      headers: {
        'Access-Control-Allow-Origin': '*',
        'Access-Control-Allow-Credentials': true,
      },
      body: "Internal Server Error Oops: " + err
    }
  }
}
